### ISSUE TITLE
Parallel test execution leads to `ConcurrentModificationException`

### ISSUE THREAD

**Issue by elgreco247** [lsoares/selenium-testing-library#5]
Parallel test execution leads to `ConcurrentModificationException`

```
 java.util.ConcurrentModificationException
	at java.base/java.util.HashMap.computeIfAbsent(HashMap.java:1221)
	at seleniumtestinglib.JSExecutorKt.loadScript(JSExecutor.kt:22)
	at seleniumtestinglib.JSExecutorKt.ensureScript(JSExecutor.kt:9)
	at seleniumtestinglib.JSExecutorKt.ensureScript$default(JSExecutor.kt:6)
	at seleniumtestinglib.queries.ApiKt.executeTLScript(Api.kt:68)
	at seleniumtestinglib.queries.ApiKt.access$executeTLScript(Api.kt:1)
	at seleniumtestinglib.locators.ByRole.findElements(ByRole.kt:340)
	...
```

I think you need to use a `ConcurrentMap` here: https://github.com/lsoares/selenium-testing-library/blob/main/lib/src/main/kotlin/seleniumtestinglib/JSExecutor.kt#L20

**Comment by lsoares** [comment]
can you check now?

**Comment by elgreco247** [comment]
Do I need to build the project locally to check? If so, then it will have to wait until I find the time for it.

**Comment by lsoares** [comment]
Are you using the library from Maven Central?
If yes no hurries. I'll tell you when I release it.

A segunda, 26/02/2024, 19:48, elgreco247 ***@***.***>
escreveu:

**Comment by lsoares** [comment]
please close this one if it's fixed

**Comment by elgreco247** [comment]
Fixed with version 3.7.8
